interface/ceed-qfunction.c

```diff
@@ -77,10 +78,73 @@ int CeedQFunctionCreateInterior(Ceed ceed, CeedInt vlength,
   ierr = CeedCalloc(strlen(focca)+1, &focca_copy); CeedChk(ierr);
   strcpy(focca_copy, focca);
   (*qf)->focca = focca_copy;
+  ierr = CeedCalloc(16, &(*qf)->inputfields); CeedChk(ierr);
+  ierr = CeedCalloc(16, &(*qf)->outputfields); CeedChk(ierr);
   ierr = ceed->QFunctionCreate(*qf); CeedChk(ierr);
   return 0;
 }
 
+/**
+  @brief Create a CeedQFunction for evaluating interior (volumetric) terms.
+
+  @param ceed       A Ceed object where the CeedQFunction will be created
+  @param vlength    Vector length.  Caller must ensure that number of quadrature
+                    points is a multiple of vlength.
+  @param spec       The specification of the desired qFunction
+  @param[out] qf    Address of the variable where the newly created
+                     CeedQFunction will be stored
+
+  @return An error code: 0 - success, otherwise - failure
+**/
+int CeedQFunctionCreateInteriorFromGallery(Ceed ceed, CeedInt vlength,
```

Comment:
We will want to do this carefully - we don't want to modify this function into a long if-else chain in the long term. I started the discussion in PR #178 on how we can do this in a maintainable way that is backend agnostic

Reply:
Since the discussion on the Gallery was idle I had to implement something, I went for the most straightforward as this is not the purpose of this PR.

Reply:
Fair enough. The interface is almost the same one I proposed, so it should be easy to swap in our agreed upon version once we make progress on that front.